// frontend/src/api/aesthetic.api.ts
import { apiClient } from './client';
import { Aesthetic, QuizAnswer, StyleQuizResult, QuizQuestion } from '@/types/aesthetic.types';

export const aestheticApi = {
    getAll: async (): Promise<Aesthetic[]> => {
        const response = await apiClient.get('/aesthetics');
        return response.data;
    },

    getById: async (id: string): Promise<Aesthetic> => {
        const response = await apiClient.get(`/aesthetics/${id}`);
        return response.data;
    },

    search: async (keyword: string): Promise<Aesthetic[]> => {
        const response = await apiClient.get('/aesthetics/search', {
            params: { keyword },
        });
        return response.data;
    },

    getQuizQuestions: async (): Promise<QuizQuestion[]> => {
        const response = await apiClient.get('/aesthetics/quiz/questions');
        return response.data;
    },

    submitQuiz: async (answers: QuizAnswer[]): Promise<StyleQuizResult> => {
        const response = await apiClient.post('/aesthetics/quiz/submit', { answers });
        return response.data;
    },
};
